modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/Joining.scala

```diff
@@ -242,6 +243,13 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
 
   private def validateHandshake(registrationRequest: RegistrationRequest, remoteAddress: Option[Host]): F[Unit] =
     for {
+
+      versionHash <- BuildInfo.version.hashF
+      _ <-
+        if (registrationRequest.version === versionHash)
```

Comment:
Maybe a tiny byt shorter:

```scala
VersionMismatch.raiseError[F, Unit].whenA(registrationRequest.version =!= versionHash)
```

Reply:
Should I change all the conditions in Joining to match that?

Reply:
Not necessarily. Generally, I think when we figure out some better way of doing something we can introduce it gradually (when we change or refactor older code).